### DISCUSSION TITLE
Voxel metrics summarized into pixels

### DISCUSSION THREAD

**Discussion by RCBlackburn** [r-lidar/lidR#591] [Q&A]
Does the package currently allow for a workaround of summarizing the `voxel_metrics()` output by using `pixel_metrics()`? For example, in the below code, I am attempting to summarize mean points per voxel with 20 m resolution pixels using the Megaplot.laz file. However, `voxel_metrics()` needs to be inside the custom function `std_voxel()` and requires an las which `pixel_metrics()` can't seem to supply. This results in the entire las file being processed at once with `pixel_metrics()` output having one value that is the mean of the entire las. Essentially, I was wondering if there is a way to use `voxel_metrics()` within `pixel_metrics()` currently or if it could be easily adjusted to work? 

```r
## function to count points in voxels created by lidR::voxel_metrics()
vox_mt <- function(z)
{

  metrics =list(
    SVi = length(z)
  )
  return(metrics)
  }

## function to summarize voxel metrics to an las
std_voxel <- function(las, res){
  vox <- lidR::voxel_metrics(las, func = vox_mt(Z), res = res, all_voxels = TRUE)

  fullvox <- as.data.frame(vox)
  voxel_summ <- fullvox %>% summarise(SVi_mean  = mean(SVi, na.rm = T))

  return(voxel_summ)
}

## read in example plot from lidR package
LASfile <- system.file("extdata", "Megaplot.laz", package="lidR")
las <- readLAS(LASfile)

## clip center to make a plot size las
center <- st_buffer(st_as_sf(st_centroid(st_as_sfc(st_bbox(las)))), 11.34)

las_plot <- clip_roi(las, center)

## run function on a plot
std_voxel(las = las_plot, res = 2)

## run function across larger las at a defined pixel resolution
v_metrics <- pixel_metrics(las, func = ~std_voxel(las, res = 2), res = 20)

## plot output
plot(v_metrics)

```

**Comment by Jean-Romain** [comment]
It is unclear what you are trying to do. To my understanding your are trying to count the number of points per 2x2x2 voxel and then average that per 20x20 pixel.

I think this should do the job

```r
library(lidR)
LASfile <- system.file("extdata", "Megaplot.laz", package="lidR")
las <- readLAS(LASfile)

vox = voxel_metrics(las, .(N = length(Z)), res = 2)
vox = LAS(vox)              
pix = pixel_metrics(vox, .(avgN = mean(N)), res = 20)
```
